require.js

```diff
@@ -1119,12 +1149,46 @@
     };
 
 
+    Require.parseMJSONDependencies = function parseMJSONDependencies(jsonRoot) {
+
+        // Clear commented require calls
+        var rootEntries = Object.keys(jsonRoot),
+            i=0, iLabel, dependencies = [], iLabelObject;
+        while ((iLabel = rootEntries[i])) {
+            iLabelObject = jsonRoot[iLabel];
+            if(iLabelObject.hasOwnProperty("prototype")) {
+                dependencies.push(iLabelObject["prototype"]);
+            }
+            else if(iLabelObject.hasOwnProperty("object")) {
+                dependencies.push(iLabelObject["object"]);
+            }
+            i++;
+        }
+        return dependencies;
+    };
+
+
     // Built-in compiler/preprocessor "middleware":
 
     Require.DependenciesCompiler = function(config, compile) {
         return function(module) {
             if (!module.dependencies && module.text !== void 0) {
-                module.dependencies = config.parseDependencies(module.text);
+                if (module.location && (endsWith(module.location, ".meta") || endsWith(module.location, ".mjson"))) {
+
+                    if (typeof module.exports !== "object" && typeof module.text === "string") {
+                        module.parsedText = JSON.parse(module.text);
+                        if (module.parsedText.montageObject) {
+                            throw new Error(
+                                'using reserved word as property name, \'montageObject\' at: ' +
+                                module.location
+                            );
+                        }
+                    }
+                    module.dependencies = Require.parseMJSONDependencies(module.parsedText);
```

Comment:
Maybe MJSON dependency parsing should be its own compiler?

Reply:
It could, and be introduced in the compiler chain, not sure it's worth it? If so the current one would have to be renamed something like JavaScriptDependenciesCompiler

Reply:
Fair enough. My two cents is javascript compilation should be done baseline by mr, and loaders/compilers should just be for montage to add support for non-standard modules. This is fine for now.